apps/api/src/database/schema/trading/defs.ts

```diff
@@ -17,105 +13,36 @@ import {
   varchar,
 } from "drizzle-orm/pg-core";
 
-export const crossChainTradingType = pgEnum("cross_chain_trading_type", [
-  "disallowAll",
-  "disallowXParent",
-  "allow",
-]);
+import { competitions, teams } from "../core/defs.js";
 
-export const teams = pgTable(
-  "teams",
-  {
-    id: uuid().primaryKey().notNull(),
-    name: varchar({ length: 100 }).notNull(),
-    email: varchar({ length: 100 }).notNull(),
-    contactPerson: varchar("contact_person", { length: 100 }).notNull(),
-    apiKey: varchar("api_key", { length: 400 }).notNull(),
-    walletAddress: varchar("wallet_address", { length: 42 }),
-    bucketAddresses: text("bucket_addresses").array(),
-    metadata: jsonb(),
-    isAdmin: boolean("is_admin").default(false),
-    active: boolean().default(false),
-    deactivationReason: text("deactivation_reason"),
-    deactivationDate: timestamp("deactivation_date", {
-      withTimezone: true,
-    }),
-    createdAt: timestamp("created_at", {
-      withTimezone: true,
-    }).defaultNow(),
-    updatedAt: timestamp("updated_at", {
-      withTimezone: true,
-    }).defaultNow(),
-  },
-  (table) => [
-    index("idx_teams_active").on(table.active),
-    index("idx_teams_api_key").on(table.apiKey),
-    index("idx_teams_is_admin").on(table.isAdmin),
-    index("idx_teams_metadata_ref_name").using(
-      "btree",
-      sql`(((metadata -> 'ref'::text) ->> 'name'::text))`,
-    ),
-    unique("teams_email_key").on(table.email),
-    unique("teams_api_key_key").on(table.apiKey),
-    unique("teams_wallet_address_key").on(table.walletAddress),
-  ],
+export const tradingComps = pgSchema("trading_comps");
+
+export const crossChainTradingType = tradingComps.enum(
+  "cross_chain_trading_type",
+  ["disallowAll", "disallowXParent", "allow"],
 );
 
-export const competitions = pgTable(
-  "competitions",
+export const tradingCompetitions = tradingComps.table(
+  "trading_competitions",
   {
-    id: uuid().primaryKey().notNull(),
-    name: varchar({ length: 100 }).notNull(),
-    description: text(),
-    startDate: timestamp("start_date", { withTimezone: true }),
-    endDate: timestamp("end_date", { withTimezone: true }),
-    status: varchar({ length: 20 }).notNull(),
+    competitionId: uuid()
+      .primaryKey()
+      .references(() => competitions.id, {
+        onDelete: "cascade",
+        onUpdate: "cascade",
+      }),
```

Comment:
We "extend" the core competitions table to add columns specific to trading compeitions.